src/Game/Managers/TargetManager.cs

```diff
@@ -283,5 +283,176 @@ public static void SendMultiTarget(ushort x, ushort y, sbyte z)
             MultiTargetInfo = null;
             ClearTargetingWithoutTargetCancelPacket();
         }
+
+        public static bool IsMobileSelectableAsTarget(Mobile mobile, int type)
+        {
```

Comment:
I totally agree. I made the changes, so both methods now expect Serial instead of Mobile, get mobile from World.Mobiles and check for null. No place for mistake.